examples/with-cookie-auth/pages/api/profile.js

```diff
@@ -1,9 +1,8 @@
-const { send, createError, run } = require('micro')
-const fetch = require('isomorphic-unfetch')
+import fetch from 'isomorphic-unfetch'
 
-const profile = async (req, res) => {
+export default async (req, res) => {
   if (!('authorization' in req.headers)) {
```

Comment:
Note here, is this checking that the authorization header is not empty?, also there's a `JSON.parse` below, it can fail easily if it's empty or just an invalid json and is not inside a `try/catch`, so it may be better to have an util function that receives the header and returns the token after validating it.

Reply:
Good points 🤔 

>  it can fail easily if it's empty

If it's empty it will return 401.

> or just an invalid json

I could check the `content-type` along with the existence of the `Authorization` header.

If both criteria are met then we can proceed. I could put this validation in a function like you said, like `validateHeaders()`, What do you think?